spec/jobs/reports/idv_legacy_conversion_supplement_report_spec.rb

```diff
@@ -0,0 +1,333 @@
+                issuer: iaa2_sp2.issuer,
+                friendly_name: iaa2_sp2.friendly_name,
+                year_month: '202009',
+                year_month_readable: 'September 2020',
+                user_count: '1',
+              },
+            )
+          end
+          expect(csv.length).to eq(2)
+        end
+      end
+
+      context 'with multiple upgraded profiles for an issuer from September-October 2020' do
+        let(:partner_account3) { create(:partner_account) }
+
+        let(:iaa3_range) { DateTime.new(2020, 9, 1).utc..DateTime.new(2021, 8, 30).utc }
+        let(:expired_iaa_range) { DateTime.new(2019, 9, 1).utc..DateTime.new(2020, 8, 31).utc }
+
+        let(:gtc3) do
+          create(
+            :iaa_gtc,
+            gtc_number: 'gtc9101',
+            partner_account: partner_account3,
+            start_date: iaa3_range.begin,
+            end_date: iaa3_range.end,
+          )
+        end
+
+        let(:iaa3) { 'iaa3' }
+
+        let(:iaa3_sp1) do
+          create(
+            :service_provider,
+            iaa: iaa3,
+            iaa_start_date: iaa3_range.begin,
+            iaa_end_date: iaa3_range.end,
+          )
+        end
+
+        let(:iaa_order3) do
+          build_iaa_order(order_number: 3, date_range: iaa3_range, iaa_gtc: gtc3)
+        end
+
+        let(:iaa_order_expired) do
+          build_iaa_order(order_number: 2, date_range: expired_iaa_range, iaa_gtc: gtc3)
+        end
+
+        let(:integration_1) do
+          build_integration(
+            issuer: iaa3_sp1.issuer,
+            partner_account: partner_account3,
+          )
+        end
+
+        let(:user1) { create(:user) }
+
+        let(:user2) { create(:user) }
+
+        let(:csv) { CSV.parse(report.perform(Time.zone.today), headers: true) }
+
+        before do
+          iaa_order3.integrations << integration_1
+
+          create(
+            :sp_upgraded_biometric_profile,
+            issuer: iaa3_sp1.issuer, user_id: user1.id, upgraded_at: iaa3_range.begin + 1.day,
+          )
+          create(
+            :sp_upgraded_biometric_profile,
+            issuer: iaa3_sp1.issuer, user_id: user2.id, upgraded_at: iaa3_range.begin + 1.month,
+          )
+        end
+
+        it 'finds data for the same issuer with different year_months' do
+          expect(csv.length).to eq(2)
+          aggregate_failures do
+            row = csv.find { |r| r['issuer'] == iaa3_sp1.issuer && r['year_month'] == '202009' }
+
+            expect(row.to_h.symbolize_keys).to eq(
+              {
+                iaa_order_number: 'gtc9101-0003',
+                iaa_start_date: '2020-09-01',
+                iaa_end_date: '2021-08-30',
+                issuer: iaa3_sp1.issuer,
+                friendly_name: iaa3_sp1.friendly_name,
+                year_month: '202009',
+                year_month_readable: 'September 2020',
+                user_count: '1',
+              },
+            )
```

Comment:
if we're doing `aggregate_failures` ... maybe we just do one assertion per attribute?
```suggestion
            expect(row['iaa_order_number']).to eq('gtc9101-0003')
            expect(row['iaa_start_date']).to eq('2020-09-01')
            expect(row['iaa_end_date']).to eq('2021-08-30')
            expect(row['issuer']).to eq(iaa3_sp1.issuer)
            expect(row['friendly_name']).to eq(iaa3_sp1.friendly_name)
            expect(row['year_month']).to eq('202009')
            expect(row['year_month_readable']).to eq('September 2020')
            expect(row['user_count']).to eq('1')
```

Reply:
I think of the aggregate_failures handling the loop but I don't prefer so many assertions (more aesthetically than 'one assertion rule') but it does match the codebase.